lnrpc/switchrpc/switch.proto

```diff
@@ -174,3 +179,30 @@ message TrackOnionResponse {
     // error decryption on the server and return the encrypted error blob.
     bytes encrypted_error = 4;
 }
+
+// BuildOnionRequest includes the necessary information to construct a Sphinx
+// onion packet.
+message BuildOnionRequest {
+    // The route for which the onion packet should be built.
+    lnrpc.Route route = 1;
+
+    // The payment hash associated with the HTLC.
+    bytes payment_hash = 2;
+
+    // A session key for onion packet construction, if not provided, a new one
+    // will be generated.
+    bytes session_key = 3;
+}
+
+// BuildOnionResponse contains the constructed onion packet.
+message BuildOnionResponse {
+    // The constructed onion packet in bytes.
+    bytes onion_blob = 1;
+
+    // The session key used for building the onion packet.
+    bytes session_key = 2;
```

Comment:
should be not set imo if we provide the session key in the first place. Is it still set ?